### PR TITLE
Add ExplicitBitVect prop and query

### PR DESCRIPTION
**Pull request by bp-kelley**
Adds queries for bit explicit bit vector prop.

Example:

```
    m = Chem.MolFromSmiles("CC")
    for atom in m.GetAtoms():
      bv = DataStructs.ExplicitBitVect(4)
      bv.SetBit(atom.GetIdx())
      atom.SetExplicitBitVectProp("prop", bv)
        
    sma = Chem.MolFromSmarts("C")
    for atom in sma.GetAtoms():
        bv = DataStructs.ExplicitBitVect(4)
        bv.SetBit(1)
        qa = rdqueries.HasBitVectPropWithValueQueryAtom("prop", bv, tolerance=0.0)

    res = m.GetSubstructMatches(sma)
    assertEqual(res, ((1,),))
```

This returns true if the tanimoto distance between bv and the atom bit vector prop "prop" is less than or equal to the supplied value.

### CODE REVIEW

File: Code/GraphMol/QueryOps.h [hunk 1]
```diff
@@ -922,12 +924,86 @@ class HasPropWithValueQuery<TargetPtr, std::string>
...
+      : Queries::EqualityQuery<int, TargetPtr, true>(), propname(prop), val(v), tol(tol) {
+    this->setDescription("HasPropWithValue");
+    this->setDataFunc(0);
+  };
+
+  virtual bool Match(const TargetPtr what) const {
+    bool res = what->hasProp(propname);
+    if (res) {
+      try {
+        const ExplicitBitVect &bv = what->template getProp<const ExplicitBitVect&>(propname);
```
Comment (greglandrum) [inline Code/GraphMol/QueryOps.h]: this use of `template` is, IMO, a really horrible bit of C++ syntax. Is it actually necessary here?

File: Code/GraphMol/Wrap/Queries.cpp [hunk 2]
```diff
@@ -214,6 +223,16 @@ struct queries_wrapper {
                 "value +- tolerance",
                 python::return_value_policy<python::manage_new_object>());
 
+    python::def("HasBitVectPropWithValueQueryAtom",
+                PropQueryWithTol<Atom, QueryAtom>,
+                (python::arg("propname"), python::arg("val"),
+                 python::arg("negate") = false, python::arg("tolerance") = 0),
+                "Returns a QueryAtom that matches when the propery 'propname' "
+                "has the specified explicit bit vector"
+                " value.  Tolerance here is allowed difference in Tanimoto",
```
Comment (greglandrum) [inline Code/GraphMol/Wrap/Queries.cpp]: maybe: `The tolerance is the allowed Tanimoto distance`

**Comment by greglandrum** [comment]
This opens up some interesting use cases.

File: Code/GraphMol/QueryOps.h [hunk 3]
```diff
@@ -922,12 +924,86 @@ class HasPropWithValueQuery<TargetPtr, std::string>
...
+ public:
+  HasPropWithValueQuery()
+      : Queries::EqualityQuery<int, TargetPtr, true>(), propname(), val(), tol(1.0) {
+    this->setDescription("HasPropWithValue");
+    this->setDataFunc(0);
+  };
+  
+  explicit HasPropWithValueQuery(const std::string &prop,
+                                 const ExplicitBitVect &v,
+                                 float tol = 1.0)
```
[the remark this answers is not in the document]
Comment (bp-kelley) [reply Code/GraphMol/QueryOps.h]: Good point.  The default the python wrapper is 0...

File: Code/GraphMol/QueryOps.h (hunk 1 above)
Comment (bp-kelley) [reply Code/GraphMol/QueryOps.h]: I think this was left over from a really old compiler.  I’m happy to take it out.

Comment (greglandrum) [reply Code/GraphMol/QueryOps.h]: Unlucky... ah well, guess we're stuck with it

Comment (bp-kelley) [reply Code/GraphMol/QueryOps.h]: I think this happens because `what` is also a template and during the first pass, the compiler doesn't know the type.

**Pull request merged**